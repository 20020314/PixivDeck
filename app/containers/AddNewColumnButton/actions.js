// @flow
import type { User } from '../../types/user'
import type { Action } from './actionTypes.js'
import { ADD_USER_ILLUST } from './constants'

export function addUserIllust(user: User): Action {
  return {
    type: ADD_USER_ILLUST,
    user,
  }
}
