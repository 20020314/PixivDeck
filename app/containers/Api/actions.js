// @flow
import type { Response } from 'services/api'

import { API_REQUEST_SUCCESS } from './actionTypes'
import type { ApiRequestSuccess } from './actionTypes'

export function apiRequestSuccess(response: Response): ApiRequestSuccess {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
