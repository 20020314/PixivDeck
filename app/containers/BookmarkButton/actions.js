// @flow
import type { Restrict } from './types'

import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILURE,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_FAILURE,
} from './actionTypes'
import type {
  AddBookmarkRequest,
  AddBookmarkSuccess,
  AddBookmarkFailure,
  DeleteBookmarkRequest,
  DeleteBookmarkSuccess,
  DeleteBookmarkFailure,
} from './actionTypes'

export function addBookmarkRequest(
  id: number,
  restrict: Restrict
): AddBookmarkRequest {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}
export function addBookmarkSuccess(
  id: number,
  restrict: Restrict
): AddBookmarkSuccess {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}
export function addBookmarkFailure(
  id: number,
  error: string
): AddBookmarkFailure {
  return {
    type: ADD_BOOKMARK_FAILURE,
    id,
    error,
  }
}
export function deleteBookmarkRequest(id: number): DeleteBookmarkRequest {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}
export function deleteBookmarkSuccess(id: number): DeleteBookmarkSuccess {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}
export function deleteBookmarkFailure(
  id: number,
  error: string
): DeleteBookmarkFailure {
  return {
    type: DELETE_BOOKMARK_FAILURE,
    id,
    error,
  }
}
