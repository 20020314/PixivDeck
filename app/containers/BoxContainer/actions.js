// @flow
import type { Action } from './actionTypes.js'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
