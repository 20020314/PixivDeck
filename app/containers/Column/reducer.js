// @flow
import update from 'utils/update'

export type BaseColumn = {
  illustIds: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    default:
      return state
  }
}
