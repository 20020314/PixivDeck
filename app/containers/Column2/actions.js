// @flow
import type { Action } from './actionTypes.js'
import { ADD_COLUMN } from './constants'
import type { Body } from './reducer'

export function addColumn(id: string, body: Body): Action {
  return {
    type: ADD_COLUMN,
    id,
    body,
  }
}
