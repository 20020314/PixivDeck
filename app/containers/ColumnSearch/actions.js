// @flow
import type { Action } from './actionTypes.js'
import {
  ADD_COLUMN,
  ADD_COLUMN_SUCCESS,
  FETCH,
  FETCH_NEW,
  FETCH_NEXT,
  SET_NEXT_URL,
  FETCH_SUCCESS,
  FETCH_NEXT_SUCCESS,
  FETCH_NEW_SUCCESS,
  SET_MIN_BOOKBOOK,
  SET_INTERVAL,
  FETCH_FAILRE,
  FETCH_NEXT_FAILRE,
  FETCH_NEW_FAILRE,
} from './constants'
import type { Response } from 'services/api'
import type { ColumnId } from './reducer'

export function addColumn(id: ColumnId): Action {
  return {
    type: ADD_COLUMN,
    id,
  }
}

export function addColumnSuccess(id: ColumnId): Action {
  return {
    type: ADD_COLUMN_SUCCESS,
    id,
  }
}

export function fetch(id: ColumnId): Action {
  return {
    type: FETCH,
    id,
  }
}

export function fetchNew(id: ColumnId): Action {
  return {
    type: FETCH_NEW,
    id,
  }
}

export function fetchNext(id: ColumnId): Action {
  return {
    type: FETCH_NEXT,
    id,
  }
}

export function setNextUrl(id: ColumnId, nextUrl: string): Action {
  return {
    type: SET_NEXT_URL,
    id,
    nextUrl,
  }
}

export function fetchSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<number>
): Action {
  return {
    type: FETCH_SUCCESS,
    id,
    response,
    ids,
  }
}

export function fetchNextSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<number>
): Action {
  return {
    type: FETCH_NEXT_SUCCESS,
    id,
    response,
    ids,
  }
}

export function fetchNewSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<number>
): Action {
  return {
    type: FETCH_NEW_SUCCESS,
    id,
    response,
    ids,
  }
}

export function setMinBookbook(id: ColumnId, minBookmarks: number): Action {
  return {
    type: SET_MIN_BOOKBOOK,
    id,
    minBookmarks,
  }
}

export function setInterval(id: ColumnId, interval: number): Action {
  return {
    type: SET_INTERVAL,
    id,
    interval,
  }
}

export function fetchFailre(id: ColumnId, error: string): Action {
  return {
    type: FETCH_FAILRE,
    id,
    error,
  }
}

export function fetchNextFailre(id: ColumnId, error: string): Action {
  return {
    type: FETCH_NEXT_FAILRE,
    id,
    error,
  }
}

export function fetchNewFailre(id: ColumnId, error: string): Action {
  return {
    type: FETCH_NEW_FAILRE,
    id,
    error,
  }
}
