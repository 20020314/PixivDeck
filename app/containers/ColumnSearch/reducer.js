// @flow
import ms from 'ms'
import update from 'utils/update'
import { handleRehydrate } from 'utils/handleReydrate'
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type ColumnId = string

export type ColumnSearch = {
  illustIds: Array<number>,
  nextUrl: ?string,
  minBookmarks: number,
  interval: number,
}

export type State = $Shape<{ [ColumnId]: ColumnSearch }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_COLUMN_SUCCESS:
      return update(state, action, {
        illustIds: [],
        nextUrl: null,
        minBookmarks: 0,
        interval: ms('1m'),
      })

    case Actions.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })

    case Actions.SET_INTERVAL:
      return update(state, action, { interval: action.interval })

    case Actions.FETCH_SUCCESS:
    case Actions.FETCH_NEXT_SUCCESS:
    case Actions.FETCH_NEW_SUCCESS:
      return update(state, action, { illustIds: action.ids })

    case Actions.SET_MIN_BOOKBOOK:
      return update(state, action, { minBookmarks: action.minBookmarks })

    case REHYDRATE:
      return handleRehydrate(state, action, 'ColumnSearch')

    default:
      return state
  }
}
