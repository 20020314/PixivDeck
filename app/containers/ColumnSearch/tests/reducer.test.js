// @flow
import reducer, { initialState } from '../reducer'
import * as actions from '../actions'

const defaultState = {
  ids: [1, 2, 3],
  nextUrl: 'dummy',
  interval: 100,
  minBookmarks: 100,
  usersIn: 0,
}

test('default action', () => {
  // $FlowFixMe
  expect(reducer(initialState, { type: 'default action' })).toMatchSnapshot()
})

test('setUsersIn', () => {
  expect(
    reducer(initialState, actions.setUsersIn('fake', 100))
  ).toMatchSnapshot()
})

test('reset ids', () => {
  expect(
    reducer({ fate: defaultState }, actions.resetIds('fate'))
  ).toMatchSnapshot()
})
