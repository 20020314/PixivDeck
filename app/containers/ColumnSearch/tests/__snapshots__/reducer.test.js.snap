// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default action 1`] = `Object {}`;

exports[`reset ids 1`] = `
Object {
  "fate": Object {
    "ids": Array [],
    "interval": 100,
    "minBookmarks": 100,
    "nextUrl": "dummy",
    "usersIn": 0,
  },
}
`;

exports[`setUsersIn 1`] = `
Object {
  "fake": Object {
    "usersIn": 100,
  },
}
`;
